corehq/apps/geospatial/reports.py

```diff
@@ -277,6 +278,22 @@ def _aggregate_query(self, query):
             precision = find_precision(query, case_property)
         return apply_geohash_agg(query, case_property, precision)
 
+    @property
+    def features_from_request(self):
+        features = {}
+        features_json = self.request.GET.get('features')
+        if features_json:
+            try:
+                features |= json.loads(features_json)
+            except json.JSONDecodeError:
+                raise ValueError(f'{features_json!r} parameter is not valid JSON')
+        selected_feature_id = self.request.GET.get('selected_feature_id')
```

Comment:
nit: how about extracting a method `_expand_geo_polygon`?
I am not sure if editing the feature is the right thing to do here. If someone refers to `features_from_request`, they would assume its the one that is exactly what was set in the request.

Reply:
> how about extracting a method _expand_geo_polygon?

Sure, I could alternatively extract this to a `saved_polygon_from_request` function. That way we can make it more explicit which features are coming from the request JSON and which are being retrieved from the back-end. What are your thoughts?

> I am not sure if editing the feature is the right thing to do here.

Just noting that we're not explicitly editing the saved features here but rather just parsing them into a dict format with randomly generated IDs (saved features don't have an ID). This is being done to keep it in the same format as the dict of features coming from the front-end.

Reply:
`saved_polygon_from_request` sounds good to me.

> Just noting that we're not explicitly editing the saved features 

Right. I just meant to say the value of features coming in the request.

Reply:
[c9476db](https://github.com/dimagi/commcare-hq/pull/34511/commits/c9476db4fa2369002a2ccc78be1eee8e97c18822)

Reply:
Thanks @zandre-eng 

nit: Should the name be `selected_polygon_id` of the variable and of the request param? or it could be anything other than a polygon as well?